contrib/envconfig/client_config_load.go

```diff
@@ -0,0 +1,412 @@
+	// options structure.
+	DisableEnv bool
+
+	// If true and a codec is configured, the data converter of the client will point to the codec remotely. Users
+	// should usually not set this and rather configure the codec locally. Users should especially not enable this for
+	// clients used by workers since they call the codec repeatedly even during workflow replay.
+	IncludeRemoteCodec bool
+
+	// Override the environment variable lookup. If nil, defaults to [EnvLookupOS].
+	EnvLookup EnvLookup
+}
+
+// LoadClientOptions loads client options from file and then applies environment variable overrides. This will not fail
+// if the config file does not exist. This is effectively a shortcut for [LoadClientConfigProfile] +
+// [ClientConfigProfile.ToClientOptions]. See [LoadClientOptionsOptions] and [ClientConfigProfile] on how files and
+// environment variables are applied.
+//
+// WARNING: Environment configuration is currently experimental.
+func LoadClientOptions(options LoadClientOptionsOptions) (client.Options, error) {
+	// Load profile
+	prof, err := LoadClientConfigProfile(LoadClientConfigProfileOptions{
+		ConfigFilePath:    options.ConfigFilePath,
+		ConfigFileData:    options.ConfigFileData,
+		ConfigFileProfile: options.ConfigFileProfile,
+		ConfigFileStrict:  options.ConfigFileStrict,
+		DisableFile:       options.DisableFile,
+		DisableEnv:        options.DisableEnv,
+		EnvLookup:         options.EnvLookup,
+	})
+	if err != nil {
+		return client.Options{}, err
+	}
+
+	// Convert to client options
+	return prof.ToClientOptions(ToClientOptionsOptions{IncludeRemoteCodec: options.IncludeRemoteCodec})
+}
+
+// [LoadClientConfigOptions] are options for [LoadClientConfig].
+type LoadClientConfigOptions struct {
+	// Override the file path to use to load the TOML file for config. Defaults to TEMPORAL_CONFIG_FILE environment
+	// variable or if that is unset/empty, defaults to [os.UserConfigDir]/temporal/temporal.toml. If ConfigFileData is
+	// set, this cannot be set and no file loading from disk occurs.
+	ConfigFilePath string
+
+	// TOML data to load for config. If set, this overrides any file loading. Cannot be set if ConfigFilePath is set.
+	ConfigFileData []byte
+
+	// If true, will error if there are unrecognized keys.
+	ConfigFileStrict bool
+
+	// Override the environment variable lookup (only used to determine which config file to load). If nil,
+	// defaults to [EnvLookupOS].
+	EnvLookup EnvLookup
+}
+
+// LoadClientConfig loads the client configuration structure from TOML. Does not load values from environment variables
+// (but may use environment variables to get which config file to load). This will not fail if the file does not exist.
+// See [ClientConfig.FromTOML] for details on format.
+//
+// WARNING: Environment configuration is currently experimental.
+func LoadClientConfig(options LoadClientConfigOptions) (ClientConfig, error) {
```

Comment:
This and `LoadClientOptions` don't have distinct enough names, IMO.

Maybe `LoadClientOptionsFromFile` ?

Reply:
I think it's distinct enough given its return type and such, but I can change and change the param structure to `LoadClientConfigFromFileOptions`. Would like to wait on other opinions if anyone has any.